src/devnet-utils.ts

```diff
@@ -56,94 +57,125 @@ export class DevnetUtils implements Devnet {
         return `${this.hre.starknet.networkConfig.url}`;
     }
 
-    private async withErrorHandler<T>(asyncFn: () => Promise<T>, errorMessage: string) {
+    private async requestHandler(
+        url: string,
+        method: Method,
+        errorMessage?: string,
+        data?: unknown,
+        headers?: any
+    ): Promise<AxiosResponse> {
         try {
-            return await asyncFn();
+            // Make the request
+            const axiosInstance = axios.create({
```

Comment:
Could this be a private property?

Reply:
Marking as resolved as I see this was also covered in https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/274/commits/cad87ed19051704fb1e0dc240df16f8583ab878e.